/*
 * Copyright (c) Huawei Technologies Co., Ltd. 2026-2026. All rights reserved.
 */

package com.openjiuwen.studio.agent.runtime.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

import com.openjiuwen.studio.agent.common.dto.ImportRes;
import io.swagger.annotations.ApiModel;
import jakarta.validation.Valid;
import jakarta.validation.constraints.Size;

import org.springframework.validation.annotation.Validated;

import java.io.Serializable;
import java.util.List;
import java.util.Objects;

/**
 * 导入插件响应体
 */
@ApiModel(description = "导入插件响应体")

@Validated

public class ImportRsp implements Serializable {
    private static final long serialVersionUID = 1L;

    @JsonProperty("succeed_len")
    private Integer succeedLen = 0;

    @JsonProperty("count")
    private Integer count = 0;

    @JsonProperty("failed_len")
    private Integer failedLen = 0;

    @JsonProperty("inner_plugins_msg")
    @Valid
    @Size()
    private List<PluginsMsg> innerPluginsMsg = null;

    @JsonProperty("auth_plugins_msg")
    @Valid
    @Size()
    private List<PluginsMsg> authPluginsMsg = null;

    @JsonProperty("import_list")
    @Valid
    @Size()
    private List<ImportRes> importList = null;

    public Integer getSucceedLen() {
        return succeedLen;
    }

    public ImportRsp setSucceedLen(Integer succeedLen) {
        this.succeedLen = succeedLen;
        return this;
    }

    public Integer getCount() {
        return count;
    }

    public ImportRsp setCount(Integer count) {
        this.count = count;
        return this;
    }

    public Integer getFailedLen() {
        return failedLen;
    }

    public ImportRsp setFailedLen(Integer failedLen) {
        this.failedLen = failedLen;
        return this;
    }

    public List<PluginsMsg> getInnerPluginsMsg() {
        return innerPluginsMsg;
    }

    public ImportRsp setInnerPluginsMsg(List<PluginsMsg> innerPluginsMsg) {
        this.innerPluginsMsg = innerPluginsMsg;
        return this;
    }

    public List<PluginsMsg> getAuthPluginsMsg() {
        return authPluginsMsg;
    }

    public ImportRsp setAuthPluginsMsg(List<PluginsMsg> authPluginsMsg) {
        this.authPluginsMsg = authPluginsMsg;
        return this;
    }

    public List<ImportRes> getImportList() {
        return importList;
    }

    public ImportRsp setImportList(List<ImportRes> importList) {
        this.importList = importList;
        return this;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("class ImportRsp {\n");

        sb.append("    succeedLen: ").append(toIndentedString(succeedLen)).append("\n");
        sb.append("    count: ").append(toIndentedString(count)).append("\n");
        sb.append("    failedLen: ").append(toIndentedString(failedLen)).append("\n");
        sb.append("    innerPluginsMsg: ").append(toIndentedString(innerPluginsMsg)).append("\n");
        sb.append("    authPluginsMsg: ").append(toIndentedString(authPluginsMsg)).append("\n");
        sb.append("    importList: ").append(toIndentedString(importList)).append("\n");
        sb.append("}");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        ImportRsp importRsp = (ImportRsp) o;
        return Objects.equals(this.succeedLen, importRsp.succeedLen) && Objects.equals(this.count, importRsp.count)
            && Objects.equals(this.failedLen, importRsp.failedLen) && Objects.equals(this.innerPluginsMsg,
            importRsp.innerPluginsMsg) && Objects.equals(this.authPluginsMsg, importRsp.authPluginsMsg)
            && Objects.equals(this.importList, importRsp.importList);
    }

    @Override
    public int hashCode() {
        return Objects.hash(succeedLen, count, failedLen, innerPluginsMsg, authPluginsMsg, importList);
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces
     * (except the first line).
     */
    private String toIndentedString(Object o) {
        if (o == null) {
            return "null";
        }
        return o.toString().replace("\n", "\n    ");
    }
}
